transactions.go

```diff
@@ -40,29 +40,61 @@ func (t *Transaction) ParsedAmount() (*money.Money, error) {
 
 // TransactionFilters are options to pass into the request for transactions.
 type TransactionFilters struct {
-	TagID           int64  `json:"tag_id"`
-	RecurringID     int64  `json:"recurring_id"`
-	PlaidAccountID  int64  `json:"plaid_account_id"`
-	CategoryID      int64  `json:"category_id"`
-	AssetID         int64  `json:"asset_id"`
-	Offset          int64  `json:"offset"`
-	Limit           int64  `json:"limit"`
-	StartDate       string `json:"start_date" validate:"datetime=2006-01-02"`
-	EndDate         string `json:"end_date" validate:"datetime=2006-01-02"`
-	DebitAsNegative bool   `json:"debit_as_negative"`
+	TagID           *int64  `json:"tag_id"`
+	RecurringID     *int64  `json:"recurring_id"`
+	PlaidAccountID  *int64  `json:"plaid_account_id"`
+	CategoryID      *int64  `json:"category_id"`
+	AssetID         *int64  `json:"asset_id"`
+	Offset          *int64  `json:"offset"`
+	Limit           *int64  `json:"limit"`
+	StartDate       *string `json:"start_date" validate:"omitempty,datetime=2006-01-02"`
+	EndDate         *string `json:"end_date" validate:"omitempty,datetime=2006-01-02"`
+	DebitAsNegative *bool   `json:"debit_as_negative"`
 }
 
 // ToMap converts the filters to a string map to be sent with the request as
-// GET parameters.
+// GET parameters. If the field is nil, it will not be included in the map.
+// This is useful for the query parameters in the request.
 func (r *TransactionFilters) ToMap() (map[string]string, error) {
 	ret := map[string]string{}
-	b, err := json.Marshal(r)
-	if err != nil {
-		return nil, err
+	if r.TagID != nil {
+		ret["tag_id"] = fmt.Sprintf("%d", *r.TagID)
 	}
 
-	if err := json.Unmarshal(b, &ret); err != nil {
-		return nil, err
+	if r.RecurringID != nil {
+		ret["recurring_id"] = fmt.Sprintf("%d", *r.RecurringID)
+	}
+
+	if r.PlaidAccountID != nil {
+		ret["plaid_account_id"] = fmt.Sprintf("%d", *r.PlaidAccountID)
+	}
+
+	if r.CategoryID != nil {
+		ret["category_id"] = fmt.Sprintf("%d", *r.CategoryID)
+	}
+
+	if r.AssetID != nil {
+		ret["asset_id"] = fmt.Sprintf("%d", *r.AssetID)
+	}
+
+	if r.Offset != nil {
+		ret["offset"] = fmt.Sprintf("%d", *r.Offset)
+	}
+
+	if r.Limit != nil {
+		ret["limit"] = fmt.Sprintf("%d", *r.Limit)
+	}
+
+	if r.StartDate != nil {
+		ret["start_date"] = *r.StartDate
+	}
+
+	if r.EndDate != nil {
+		ret["end_date"] = *r.EndDate
+	}
+
+	if r.DebitAsNegative != nil {
+		ret["debit_as_negative"] = fmt.Sprintf("%t", *r.DebitAsNegative)
 	}
```

Comment:
I feel like there's gotta be a better way to do this, but nothing comes to mind, so 🤷